sycl/doc/extensions/proposed/sycl_ext_oneapi_cuda_cluster_group.asciidoc

```diff
@@ -0,0 +1,326 @@
+          sycl_ext_oneapi_enqueue_functions]
+* link:../experimental/sycl/sycl_ext_oneapi_properties.asciidoc[
+    sycl_ext_oneapi_properties
+]
+
+
+== Status
+
+This is an experimental extension specification for the `ext_oneapi_cuda`
+backend, intended to provide early access to features and gather community
+feedback.  
+Interfaces defined in this specification are implemented in {dpcpp}, but they
+are not finalized and may change incompatibly in future versions of {dpcpp}
+without prior notice. *Shipping software products should not rely on APIs
+defined in this specification.*
+
+=== Backend support status 
+
+This extension is only supported by the `ext_oneapi_cuda` backend.
+
+
+== Glossary
+
+* Compute Capability: Abbreviated as "cc", a number assigned to each generation
+of NVIDIA's GPUs conveying the feature set associated with that number.
+
+
+
+== Overview
+
+NVIDIA’s compute capability (cc) 9.0 devices introduced a new level in the
+thread hierarchy, called as thread block clusters, in CUDA terminology. A thread
+block cluster, is a collection of thread blocks (a work-group in SYCL
+terminology). The work-groups which make up a cluster have the ability to access
+each other's local memory, and can be synchronized. This has various
+applications, convolutions, GEMMs and FFTs to name a few.
+
+This proposal introduces a SYCL equivalent of the same, to be called as
+`cluster_group`, and adds the methods to launch a kernel with a cluster range
+and accessing the various id's associated with the cluster
+launch in device code.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_CUDA_CLUSTER_GROUP` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not version-ed, so the
+ feature-test macro always has this value.
+|===
+
+
+=== Launching a kernel with a `cluster_group`
+
+Because of the special scheduling guarantees associated with a cluster launch,
+the backend must know which kernel would be using this feature. Thus, this 
+proposal introduces a new launch property called as `cluster_size` that will
+contain the cluster  in terms of number of workgroups.
+ 
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental::cuda {
+/**
+* Dim Dimensionality of the launch
+* size sycl::range specifying the number of workgroups in the cluster
+  in each dimension.
+*/
+template <int Dim>
+struct cluster_size {
+  cluster_size(const sycl::range<Dim>& size) : 
+    size(size) {}
+  size_t operator[](int i) { return size[i]; }
```

Comment:
```suggestion
  size_t operator[](int i)
```
We don't need a definition, just the declaration.

Reply:
Fixed, 
though I have changed this query, instead of returning the cluster size for a particular dimension, it now returns the sycl::range itself